controller/controller_server/csi_controller_server.py

```diff
@@ -590,11 +589,11 @@ def _get_object_name_and_prefix(self, request, max_name_prefix_length, max_name_
                         max_name_prefix_length
                     )
                 )
-            full_name = self._join_object_prefix_with_name(prefix, name)
-        if len(full_name) > max_name_length:
+            final_name = self._join_object_prefix_with_name(prefix, name)
+        if len(final_name) > max_name_length:
             hashed_name = utils.hash_string(name)
-            full_name = self._join_object_prefix_with_name(prefix, hashed_name)
-        return full_name[:max_name_length], prefix
+            final_name = self._join_object_prefix_with_name(prefix, hashed_name)
+        return final_name[:max_name_length]
```

Comment:
actually, I think full is more accurate in this context, since this slicing makes a final version from the full version

Reply:
done